src/EnergyPlus/HeatBalFiniteDiffManager.cc

```diff
@@ -884,88 +894,119 @@ namespace HeatBalFiniteDiffManager {
-                                             SurfaceFD(SurfNum).specHeatMaterialActuators(mat).isActuated,
-                                             SurfaceFD(SurfNum).specHeatMaterialActuators(mat).actuatedValue);
+                                             SurfaceFD(SurfNum).specHeatMaterialActuators(lay).isActuated,
+                                             SurfaceFD(SurfNum).specHeatMaterialActuators(lay).actuatedValue);
+            }
+
+            // Setup EMS Actuator and Output Variables for Heat Flux
+            // Only setup for layers 1 to N-1
+            for (int lay = 1; lay < state.dataConstruction->Construct(ConstrNum).TotLayers; ++lay) {
+                EnergyPlus::SetupEMSActuator(state,
+                                             "CondFD Surface Material Layer",
+                                             SurfaceFD(SurfNum).heatSourceFluxMaterialActuators(lay).actuatorName,
+                                             "Heat Flux",
+                                             "[W/m2]",
+                                             SurfaceFD(SurfNum).heatSourceFluxMaterialActuators(lay).isActuated,
+                                             SurfaceFD(SurfNum).heatSourceFluxMaterialActuators(lay).actuatedValue);
+                SetupOutputVariable(state,
+                                    format("CondFD Heat Source Power After Layer {}", lay),
+                                    OutputProcessor::Unit::W,
+                                    SurfaceFD(SurfNum).heatSourceFluxLayerReport(lay),
+                                    OutputProcessor::SOVTimeStepType::Zone,
+                                    OutputProcessor::SOVStoreType::State,
+                                    state.dataSurface->Surface(SurfNum).Name);
+                SetupOutputVariable(state,
+                                    format("CondFD Heat Source Energy After Layer {}", lay),
+                                    OutputProcessor::Unit::J,
+                                    SurfaceFD(SurfNum).heatSourceFluxEnergyLayerReport(lay),
+                                    OutputProcessor::SOVTimeStepType::Zone,
+                                    OutputProcessor::SOVStoreType::Summed,
+                                    state.dataSurface->Surface(SurfNum).Name,
+                                    _,
+                                    "Electricity",
+                                    "Heating",
+                                    _,
+                                    "Building");
             }
 
             TotNodes = ConstructFD(state.dataSurface->Surface(SurfNum).Construction).TotNodes; // Full size nodes, start with outside face.
-            for (Lay = 1; Lay <= TotNodes + 1; ++Lay) {                                        // include inside face node
+            for (int node = 1; node <= TotNodes + 1; ++node) {                                        // include inside face node
```

Comment:
Minor refactoring. These outputs are at the node level, not the "layer" level. Just changing the indexer here to reflect that.